### ISSUE TITLE
Support Sphinx 8

### ISSUE THREAD

**Issue by jwodder** [readthedocs/sphinx_rtd_theme#1582]
Sphinx 8 was released on July 29, yet sphinx-rtd-theme requires a Sphinx version less than 8.  Please add support for Sphinx 8.

**Comment by dylanh724** [comment]
High priority, imo!

**Comment by stephanebruckert** [comment]
This is the error we are getting https://readthedocs.org/projects/spotipy/builds/25216396/

```
The conflict is caused by:
    The user requested Sphinx~=8.0.2
    sphinx-rtd-theme 2.0.0 depends on sphinx<8 and >=5
```

**Comment by dylanh724** [comment]
Related extensions updating to 8 (links from @joola):

* https://github.com/executablebooks/MyST-Parser/issues/950
* https://github.com/readthedocs/sphinx_rtd_theme/issues/1582

**Comment by Mr0grog** [comment]
> if you want to try and install from that branch and ensure things look & work well, that would be helpful

Tried this out on a project I maintain ([wayback](https://github.com/edgi-govdata-archiving/wayback)), and it seems to work great!

Built the docs with dependencies in `pyproject.toml`:

```toml
[project.optional-dependencies]
docs = [
    "sphinx ~=8.0.2",
    "sphinx_rtd_theme @ git+https://github.com/readthedocs/sphinx_rtd_theme.git@humitos/plan-3.0",

    # Probably not relevant, but these were also in use:
    "ipython ~=8.16.1",
    "numpydoc ~=1.6.0",
    "sphinx-copybutton ~=0.5.2",
]
```

Installation resolved correct versions without warnings or errors, Sphinx docs build had no errors, and output looks OK at a quick glance.

**Comment by humitos** [comment]
We released a new version [`3.0.0rc1`](https://pypi.org/project/sphinx-rtd-theme/3.0.0rc1/) that adds support for latest Sphinx versions. Please, give it a try and open an issue for any problem or feedback you want to share with us.

**Comment by dylanh724** [comment]
@humitos I really appreciate you remembering me for this! I ended up moving to PyData's Book theme -- that 3rd column was too important to me. However, I do still appreciate this theme like crazy (and use RTD as the host) <3

**Comment by Segelzwerg** [comment]
> We released a new version [`3.0.0rc1`](https://pypi.org/project/sphinx-rtd-theme/3.0.0rc1/) that adds support for latest Sphinx versions. Please, give it a try and open an issue for any problem or feedback you want to share with us.

I can confirm that my build worked with 3.0.0rc1 and sphinx 8.0.2.
https://github.com/Segelzwerg/django-dynamic-theme/pull/163
